components/InfoModal.vue

```diff
@@ -0,0 +1,91 @@
+<template>
+        <teleport to="body">
+            <div
+                ref="modal-backdrop"
+                class="fixed z-10 inset-0 overflow-y-auto bg-black bg-opacity-50"
+            >
+            <div
+                class="flex items-start justify-center min-h-screen pt-24 text-center"
+            >
+                <div
+                    class="flex justify-between bg-white rounded-lg text-left overflow-hidden shadow-xl pb-4 pl-6 pr-2 w-1/2"
+                    role="dialog"
+                    ref="modal"
+                    aria-modal="true"
+                    aria-labelledby="modal-headline"
+                >
+                    
+                    
+                  <div class="w-full max-w-4xl m-auto pt-6">
+                      <h2 class="text-center text-xl mb-4">{{showDateTitle}}</h2>
+                      <TodoItem v-for="todo in filteredTodos" :key="todo.id" :todo="todo" @toggleChanging="(e) => changeStatus(e, todo.id)" @removeItem="(e) => remove(e, todo.id)" :infoModal="true"/>
+                  </div>
+                  <button class="text-xs mt-2 bg-transparent hover:bg-red-500 text-red-700 hover:text-white border border-red-500 hover:border-transparent rounded-full h-8 w-8 pb-1 flex items-center justify-center" @click="closeModal">
+                      &#9587;
+                  </button>
+                </div>
+            </div>
+            </div>
+        </teleport>    
+</template>
+
+<script setup lang="ts">
+const emit = defineEmits<{
+  (event: 'close'): void;
+}>();
+const todosStore = useTodosStore();
+const todos = ref(todosStore.todos);
+const props = defineProps({
```

Comment:
When using typescript it is better to use generic prop definition like 
```
const props = defineProps<{
  date: Date
  showDateTitle: string
}>()
```
since you can reuse the interfaces and types then